http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -130,12 +130,19 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
                                                                  kubernetesNamespace
       )
 
+      maybeCromwellRunnerDatabaseNames <- mayCreateCromwellRunnerDatabases(app, params.workspaceId, kubernetesNamespace)
+
       // Determine which type of identity to link to the app: pod identity, workload identity, or nothing.
-      identityType = (maybeKsaFromDatabaseCreation, app.samResourceId.resourceType, maybeCromwellDatabaseNames) match {
-        case (Some(_), _, _)                      => WorkloadIdentity
-        case (None, SamResourceType.SharedApp, _) => NoIdentity
-        case (None, _, Some(_))                   => WorkloadIdentity
-        case (None, _, _)                         => PodIdentity
+      identityType = (maybeKsaFromDatabaseCreation,
```

Comment:
Hopefully when we adopt https://github.com/DataBiosphere/terra-workspace-manager/pull/1430 we can get rid of this logic, and just use WorkloadIdentity across the board.